lib/cache.js

```diff
@@ -11,6 +11,9 @@ export const cache = atom(
     queries: {},
     refs: {},
     index: {},
+    lookups: {
+      serviceNamesByQueryId: {},
```

Comment:
I wonder if we could change `queries` to just be `.queries[queryId]` and then put `serviceName` in `.queries[queryId].serviceName`?

But then updating all queries for a given service would be more expensive, so maybe not